webapp/src/lib/components/spot-component/spot-location-create-form.svelte

```diff
@@ -5,100 +5,280 @@
+        debounceTimer = window.setTimeout(() => {
+            if (searchQuery.length > 3) {
+                searchAddress();
+            }
+        }, 500); // 0.5s delay
+    };
+
+    // Handle selection of an address from dropdown
+    const handleSelect = (location: AddressResult) => {
+        searchQuery = location.properties.formatted;
+        results = [];
+        dropdownOpen = false;
+
+        if (location.properties.country_code !== 'ca') {
+            warningModalOpen = true;
+            warningText = 'This application supports locations in Canada only!';
+        } else {
+            houseNumber = location.properties.housenumber ?? '';
+            streetAddress = `${location.properties.housenumber} ${location.properties.street}`;
+            city = location.properties.city;
+            state = location.properties.state_code;
+            countryCode = location.properties.country_code?.toUpperCase();
+            country = location.properties.country;
+            postalCode = location.properties.postcode?.replace(/\s/g, '');
+            isVerified = false; // Reset verification after selection
+            isReadOnly = false;
+        }
+    };
+
+    // Address verification using Geoapify API
+    const verifyAddress = () => {
+        const url = `https://api.geoapify.com/v1/geocode/search?text=${encodeURIComponent(
+            `${houseNumber} ${streetAddress}, ${city}, ${state}, ${postalCode}, ${country}`
+        )}&apiKey=${apiKey}`;
+
+        fetch(url)
+            .then((response) => response.json())
+            .then((result) => {
+                if (result.features.length === 0) {
+                    warningModalOpen = true;
+                    warningText = 'Address verification failed. Please check your inputs!';
+                    return;
+                }
+
+                const matchedAddress = result.features[0].properties;
+                // Validate all fields match exactly
+                const isMatch =
+                    matchedAddress.housenumber + ' ' + matchedAddress.street?.toLowerCase() ===
+                        streetAddress.toLowerCase() &&
+                    matchedAddress.city?.toLowerCase() === city.toLowerCase() &&
+                    matchedAddress.state_code?.toLowerCase() === state.toLowerCase() &&
+                    matchedAddress.postcode?.replace(/\s/g, '').toLowerCase() ===
+                        postalCode.toLowerCase();
+                if (isMatch) {
+                    isVerified = true;
+                } else {
+                    warningModalOpen = true;
+                    warningText =
+                        'Address verification failed. The entered address does not match the verified address!';
+                }
+            })
+            .catch((error) => {
+                console.log('error', error);
+                warningModalOpen = true;
+                warningText = 'An error occurred during verification. Please try again!';
```

Comment:
This verify function doesn't work for a lot of cases. I think it is due to the inconsistent between geocodio (in the backend) and geoapify (in the front end). For instances:
* 140 Portage Avenue, Winnipeg, MB R3J 2H7, Canada returned by the search result but was unverify
* 98 Portage Avenue East, Winnipeg, MB R3B 0W7, Canada is on Google map but was not accepted
* 30 Maclean Crescent is a valid address but was not accepted
I think some of these can be fixed by doing a loop to verify instead of just checking the first result [0]